tool/src/kotlin/mod.rs

```diff
@@ -861,6 +869,8 @@ retutnVal.option() ?: return null
         special_methods: &mut SpecialMethods,
         method: &'cx hir::Method,
         self_type: Option<&'cx SelfType>,
+        struct_name: Option<&str>,
+        callback_params: &mut Vec<CallbackParamInfo>,
```

Comment:
thought: these callback params could also be collected on the TyGenContext, probably